DfciPkg/DfciManager/DfciVarPolicies.c

```diff
@@ -102,6 +99,64 @@ InitializeAndSetPolicyForAllDfciVariables (
     goto Done;
   }
 
+  //
+  // Lock most variables when DfciLock variable is created.
+  //
+  for (i = 0; i < ARRAY_SIZE (gReadyToBootPolicies); i++) {
+    Status = RegisterVarStateVariablePolicy (
+               mVariablePolicy,
+               gReadyToBootPolicies[i].Namespace,
+               gReadyToBootPolicies[i].Name,
+               gReadyToBootPolicies[i].MinSize,
+               gReadyToBootPolicies[i].MaxSize,
+               gReadyToBootPolicies[i].AttributesMustHave,
+               gReadyToBootPolicies[i].AttributesCantHave,
+               &gDfciLockVariableGuid,
+               DFCI_LOCK_VAR_NAME,
+               1
+               );
+
+    if (EFI_ERROR (Status)) {
+      DEBUG ((DEBUG_ERROR, "%a %a: - RegisterVarStateVariablePolicy() ReadyToBoot[%d] returned %r!\n", _DBGMSGID_, __FUNCTION__, i, Status));
+      DEBUG ((DEBUG_ERROR, "%a %a: - Error registering %g:%s\n", _DBGMSGID_, __FUNCTION__, gReadyToBootPolicies[i].Namespace, gReadyToBootPolicies[i].Name));
+    }
+  }
+
+  //
+  // Make sure there is no existing lock variable
+  //
+  Status = gRT->SetVariable (

```

Comment:
Is there benefit to moving this before the RegisterVarStateVariablePolicy? Seems like you'd want the lock variable to be sane before you register policies against it.